### PR TITLE
[IMP] product: search product/template with ref

### PR DESCRIPTION
**Pull request by kcv-odoo**
Before this commit when user search on product with its `default_code` then user can find matching product but in product template it is not giving matching records when user want to search product template with default code because name_search was only implemented for product.

This commit add a mixin instead of duplicate same code to search records matching with `default_code` and add them for product and product template.

task-2822339

### CODE REVIEW

**Comment by Feyensv** [comment]
Please avoid marking a PR as 'ready for review' until it reaches the latest project stages (Testing BE green or RD review) to avoid pinging codeowners on a task that wasn't even tested and validated first (and therefore can still change a lot before being merged), thank you :pray:.

**Comment by vchu-odoo** [comment]
Sorry for the delay in review, could you please rebase and resolve the conflicts?

**Comment by kcv-odoo** [comment]
Hello @vchu-odoo :wave: 
Lots of changes has been done since last change so need to check validity of code and approach
Also need to test it again
So taking in back to Dev :innocent:

**Comment by kcv-odoo** [comment]
Hello @vchu-odoo :wave: 

Thank you for your suggestion :pray: 
I updated code accordingly!
Also I need to update `product_template`'s domain term before passing it for `product_product`'s name_search

**Review by vchu-odoo — APPROVED** [review]
Thank you for your changes!
LGTM :+1: 

Maybe it is worth to add a test on `_name_search` with `search_product_product` context to make sure nobody will break it again.

File: addons/website_sale/controllers/variant.py
Comment (Feyensv) [inline addons/website_sale/controllers/variant.py:18]: Since `_get_combination_info` is only used in `website_sale` now, can't we fix it directly in the method ?

```python
display_name = product_or_template.with_context(display_default_code=False).display_name
```

When we can, we should aim to target as precisely as possible the context updates (it makes it easier to understand, and ensures that it only impacts the places we want).

Comment (kcv-odoo) [reply addons/website_sale/controllers/variant.py]: Done! :)

**Comment by Feyensv** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**
